tests/main/prepare-image-validation-sets/task.yaml

```diff
@@ -0,0 +1,162 @@
+summary: Verify that we are able to use validation sets with the model assertion through prepare-image 
+
+# autopkgtest run only a subset of tests that deals with the integration
+# with the distro
+backends: [-autopkgtest]
+
+# disable the following distributions
+# ubuntu-14, lack of systemd-run
+# ubuntu-18.04-32: pc gadget from 20 is not available
+# ubuntu-20.04-arm*, because we use pc kernel and gadget.
+systems:
+- -ubuntu-14.04-*
+- -ubuntu-18.04-32
+- -ubuntu-20.04-arm-*
+
+environment:
+    ROOT: /home/test/tmp/
+    IMAGE: $ROOT/system-seed
+    GADGET: $ROOT/gadget
+    STORE_DIR: $(pwd)/fake-store-blobdir
+    STORE_ADDR: localhost:11028
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    mkdir -p "$ROOT"
+    chown test:test "$ROOT"
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    
+    "$TESTSTOOLS"/store-state teardown-fake-store "$STORE_DIR"
+    rm -rf "$ROOT"
+
+execute: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    install_snap_to_fakestore() {
+        local SNAP_NAME="$1"
+        local SNAP_REVISION="$2"
+        local SNAP_PATH
+
+        SNAP_PATH=$("$TESTSTOOLS"/snaps-state pack-local "$SNAP_NAME")
+        "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$SNAP_PATH"
+        cat > snap-"$SNAP_NAME"-decl.json <<EOF
```

Comment:
It looks like this could perhaps be accomplished using `make-snap-installable` param `extra_decl_json_file`?
If this is the case it could  perhaps make sense to create same concept for revision declaration and avoid this special function?

Reply:
Good point about supplying the snap-decl json file, I've added this. However extending `make-snap-installable` should probably be for another PR.